kornia/enhance/jpeg.py

```diff
@@ -0,0 +1,573 @@
+
+
+def _get_default_qt_y(device: Device, dtype: Dtype) -> Tensor:
+    """Generates default Quantization table of Y channel."""
+    return torch.tensor(
+        [
+            [16, 11, 10, 16, 24, 40, 51, 61],
+            [12, 12, 14, 19, 26, 58, 60, 55],
+            [14, 13, 16, 24, 40, 57, 69, 56],
+            [14, 17, 22, 29, 51, 87, 80, 62],
+            [18, 22, 37, 56, 68, 109, 103, 77],
+            [24, 35, 55, 64, 81, 104, 113, 92],
+            [49, 64, 78, 87, 103, 121, 120, 101],
+            [72, 92, 95, 98, 112, 100, 103, 99],
+        ],
+        device=device,
+        dtype=dtype,
+    )
+
+
+def _get_default_qt_c(device: Device, dtype: Dtype) -> Tensor:
+    """Generates default Quantization table of C channels."""
+    return torch.tensor(
+        [
+            [17, 18, 24, 47, 99, 99, 99, 99],
+            [18, 21, 26, 66, 99, 99, 99, 99],
+            [24, 26, 56, 99, 99, 99, 99, 99],
+            [47, 66, 99, 99, 99, 99, 99, 99],
+            [99, 99, 99, 99, 99, 99, 99, 99],
+            [99, 99, 99, 99, 99, 99, 99, 99],
+            [99, 99, 99, 99, 99, 99, 99, 99],
+            [99, 99, 99, 99, 99, 99, 99, 99],
+        ],
+        device=device,
+        dtype=dtype,
+    )
+
+
+def _patchify_8x8(input: Tensor) -> Tensor:
+    """Function extracts non-overlapping 8 x 8 patches from the given input image.
+
+    Args:
+        input (Tensor): Input image of the shape :math:`(B, H, W)`.
+
+    Returns:
+        output (Tensor): Image patchify of the shape :math:`(B, N, 8, 8)`.
+    """
+    # Get input shape
+    B, H, W = input.shape
+    # Patchify to shape [B, N, H // 8, W // 8]
+    output: Tensor = input.view(B, H // 8, 8, W // 8, 8).permute(0, 1, 3, 2, 4).reshape(B, -1, 8, 8)
+    return output
+
+
+def _unpatchify_8x8(input: Tensor, H: int, W: int) -> Tensor:
+    """Function reverses non-overlapping 8 x 8 patching.
+
+    Args:
+        input (Tensor): Input image of the shape :math:`(B, N, 8, 8)`.
+
+    Returns:
+        output (Tensor): Image patchify of the shape :math:`(B, H, W)`.
+    """
+    # Get input shape
+    B, N = input.shape[:2]
+    # Unpatch to [B, H, W]
+    output: Tensor = input.view(B, H // 8, W // 8, 8, 8).permute(0, 1, 3, 2, 4).reshape(B, H, W)
+    return output
+
+
+def _dct_8x8(input: Tensor) -> Tensor:
+    """Performs an 8 x 8 discrete cosine transform.
+
+    Args:
+        input (Tensor): Patched input tensor of the shape :math:`(B, N, 8, 8)`.
+
+    Returns:
+        output (Tensor): DCT output tensor of the shape :math:`(B, N, 8, 8)`.
+    """
+    # Get dtype and device
+    dtype: Dtype = input.dtype
+    device: Device = input.device
+    # Make DCT tensor and scaling
+    index: Tensor = torch.arange(8, dtype=dtype, device=device)
+    x, y, u, v = torch.meshgrid(index, index, index, index)
+    dct_tensor: Tensor = ((2.0 * x + 1.0) * u * torch.pi / 16.0).cos() * ((2.0 * y + 1.0) * v * torch.pi / 16.0).cos()
```

Comment:
unrelated comment, if the torch has a `pi` constant, why do we have one too? cc @edgarriba

Reply:
historically wasn't there plus i remember some conflicts with the device api. Worth to revisit (open a ticket) and alias `[torch/kornia].pi`

Reply:
`torch.pi` directly links to `math.pi` I guess...